plugins/modules/vsphere_copy.py

```diff
@@ -125,7 +145,8 @@ def main():
         datacenter=dict(required=False),
         datastore=dict(required=True),
         path=dict(required=True, aliases=['dest'], type='str'),
-        timeout=dict(default=10, type='int'))
+        timeout=dict(default=10, type='int'),
+        diskformat=dict(required=False, type='str'))
```

Comment:
I think you should restrict this to the valid choices. See https://github.com/ansible-collections/community.vmware/blob/7632a2f05e1b7ae619ac4aee8bebacce0d2183e5/plugins/modules/vmware_guest_boot_manager.py#L79-L83 and https://github.com/ansible-collections/community.vmware/blob/7632a2f05e1b7ae619ac4aee8bebacce0d2183e5/plugins/modules/vmware_guest_boot_manager.py#L400-L403 how it could look like.

Additionally, please add a [changelog fragment](https://docs.ansible.com/ansible/9/community/development_process.html#making-your-pr-merge-worthy). Shoutld be `minor_changes`, I think.

Reply:
This is a very specific moment. I did a quick search and couldn't find any options for this parameter in the API. Although I believe that this API refers to `vim.FileManager`. But I don't think it makes sense to skip PR without understanding what this code does explicitly, you need to find an enumeration with types for this parameter and add a selection of options to `agrument_spec`.

Reply:
@ihumster Good point, I didn't realize this at first. The module (and the API) is about uploading arbitrary files and not about uploading virtual disks. So it doesn't really make sense to define a disk format (could also be an ISO, a txt file or whatever).

@Klaas- What issue are you trying to fix? And, especially, do you really think your changes do fix this? At the moment, I don't see how they could...

Reply:
Hi,
There are some details in the linked issue vmware/pyvmomi#1064

Basically if you upload a stream optimized vmdk to vsan you need to specify the disk format. That is what the webui does when uploading a vmdk. So that is how I figured it out even though it's an undocumented parameter :)

Yes I am sure it fixes the issue I am having, I tested it with my modified version.

>  So it doesn't really make sense to define a disk format (could also be an ISO, a txt file or whatever)

Yes, it's an optional parameter needed if you upload stream optimized vmdks (maybe specific to vsan data stores. If you upload an iso you simply do not set that parameter.

I'll incorporate the changes requested here as soon as I get back home, I can't say what other options the parameter may have other than the one I found because it's undocumented by VMware as far as I can tell. I did open an issue in their SDK, so maybe they will have something to say

Reply:
@Klaas- I suggest waiting for a response in your issue in pyvmomi. Maybe VMware will respond.